src/robust.jl

```diff
@@ -0,0 +1,110 @@
+# Robust Statistics
+
+#############################
+#
+#   Trimming outliers
+#
+#############################
+
+# Trimmed set
+"""
+    trim(x, p=0.2)
+
+Remove a proportion `p` of the highest elements, and `p` of the lowest
+elements of vector `x` and return the result. To compute the trimmed
+mean of 'x', use `mean(trim(x, p))`; to compute the variance of this
+quantity, use `trimvar(x, p)`.
+
+
+If you want to compute
+the variance of `mean(trim(x,p))`, use `trimvar`.
+
+# Example
+```julia
+julia> trim([1,2,3,4,5], 0.2)
+ 2
+ 3
+ 4
+```
+"""
+trim(x::AbstractVector, p::Real=0.2) = trim!(copy(x), p)
+
+"""
+    trim!(x, p=0.2)
+
+A variant of `trim` that partially sorts `x` in place while removing
+the proportion `p` of the highest and lowest elements of `x`.
+"""
+function trim!(x::AbstractVector, p::Real=0.2)
+    n = length(x)
+    n > 0 || error("x can not be empty.")
+    0 <= p < 0.5 || error("p must be in 0 <= p < 0.5.")
```

Comment:
Should use explicitly typed exceptions rather than `error`. In this case as well as the line above it I'd use `throw(ArgumentError(...))` with the text.